### ISSUE TITLE
Upgrade Tiers Not Working

### ISSUE THREAD

**Issue by huneyb** [Tiviacz1337/Travelers-Backpack#635]
0. ALWAYS PROVIDE INFORMATION ABOUT MODLOADER (Fabric / Forge) AND MOD VERSION (For Example 1.19.2-8.2.10)
Fabric 1.20 version .02

1. Describe the bug 
I cannot get the upgrade tier to work on the new smithing table

2. Write steps to reproduce the bug
I put the backpack and iron upgrade into the smithing table and theres nothing in the output. 

3. What is expected behaviour 
I thought once you put the backpack and the upgrade in the table it outputs the upgrade backpack

5. Screenshots (If applicable)

**Comment by huneyb** [comment]
![Backpack not upgrading](https://github.com/Tiviacz1337/Travelers-Backpack/assets/136275623/e6e7673c-df2e-4ee4-838f-74b332c7b8b6)

**Comment by Dr-WeiAL** [comment]
![image](https://github.com/Tiviacz1337/Travelers-Backpack/assets/47446553/3edee3c7-341d-4864-8478-caa0b7dfa8f8)
You also need to put in leather
I suggest you install REI

**Comment by AdamFull** [comment]
Not working for me and i don't know why. It's not upgraded one
![image](https://github.com/Tiviacz1337/Travelers-Backpack/assets/25034970/6adcfaa8-44d4-470e-995c-1033b2eeb88d)

**Comment by creeperreaper132-droid** [comment]
Fabric 1.21.1, version 0.116.12
I can't use the backpack upgrades (Furnace variants)
I applied the upgrade to the right side of the GUI properly and it says to equip the backpack to use the upgrade, I equipped it, checked the furnace, and it didn't work. I even made a upgrade toggle keybind and waited for a result.
I expected for the items to cook in the furnace variants when I equipped the BP. I made sure to put on coal, but I can use the Craft upgrade along with the Tank upgrade.

<img width="134" height="165" alt="Image" src="https://github.com/user-attachments/assets/15aac0f8-fcd2-4002-b85c-9a3e6a567e15" />
